lib/go/thrift/ssl_server_socket.go

```diff
@@ -93,6 +93,9 @@ func (p *TSSLServerSocket) Open() error {
 }
 
 func (p *TSSLServerSocket) Addr() net.Addr {
+	if p.listener != nil {
+		return p.listener.Addr()
+	}
```

Comment:
this is a bug that it could return the wrong address (for example when constructed with port 0)